### PR TITLE
update virtio related functions doc comments

### PR DESCRIPTION
**Pull request by shuox**
1) update virtio related functions doc comments
2) also remove one unused APIs. Doc will be removed if we generate by doxygen.

Tracked-On: #1595

### CODE REVIEW

File: devicemodel/hw/pci/virtio/vhost.c [hunk 1]
```diff
@@ -581,6 +581,22 @@ vhost_set_mem_table(struct vhost_dev *vdev)
 	return 0;
 }
 
+/**
+ * @brief vhost_dev initialization.
+ *
+ * This interface is called to initialize the vhost_dev. It must be called
+ * before the actual feature negoitiation with the guest OS starts.
```
Comment (gvancuts) [inline devicemodel/hw/pci/virtio/vhost.c]: s/negoitiation/negotiation

Comment (shuox) [reply devicemodel/hw/pci/virtio/vhost.c]: OK, will fix it. Thanks for reviewing.

**Comment by junjiemao1** [comment]
@dbkinder Please help review and merge this one for API doc. Thanks!

**Comment by shuox** [comment]
@dbkinder sorry for late response. Rebase done.

**Pull request merged**